### ISSUE TITLE
All USB Peripherals go haywire

### ISSUE THREAD

**Issue by NicoSteyl** [x360ce/x360ce#1163]
Launching this causes all the other usb to constantly disconnect and reconnect. I am on a Asus b450 Tomahawk with ryzen 5 cpu and nvidia gtx 1660 ti. Same result using front and rear port usb.

**Comment by HaykelTV** [comment]
same for me , but only happen once i open the program (v4.xx) 
my mouse and keyboard disconnect and reconnect but only once 
then everything works fine until i close the program they disconnect and reconnect again 
even dho they are ticket off in map device to controller menu

**Comment by freecodecampster** [comment]
x360 Version 4.15.16 installed HID guardian and upon reboot none of my USB Bluetooth peripherals mouse keyboard would be recognised. After system restore and sfc didn't fix the problem I had to force Windows 10 into automatic repair mode by holding power button to force power off whilst booting up twice. Now I have to reinstall Windows grrr!
****

**Comment by EJocys** [comment]
Yes, that is very serious issue. Please try to uninstall HID Guardian manually:

1. Start your PC in Safe Mode with Networking
2. Download: https://github.com/x360ce/x360ce/blob/master/x360ce.App.Beta/Resources/HidGuardian.zip?raw=true
3. Extract to "HidGuardian" folder.
4. Run "HidGuardian\HidGuardian_Setup.bat"
5. Choose Option: 2 - Uninstall.

I will add code which creates "c:\ProgramData\X360CE\HidGuardian" folder automatically, and provides warning with instructions before installing HID Guardian.

I will try to find out what could cause HID Guardian to hide all devices. Please, could you run "HidGuardian_Export.bat" from the same zip, before uninstalling. It will export your HID Guardian configuration to _HidGuardian.reg.txt and _HidGuardian001.reg.txt file. Please send these files to support@x360ce.com.

Update: If keyboard or mouse doesn't work in Safe Mode then it is possible to remove HID Guardian in windows recovery environment, by loading "C:\Windows\System32\config\SYSTEM" registry hive file (Computer\HKEY_LOCAL_MACHINE\SYSTEM) with regedit.exe and removing all keys with "HidGuardian" string in them. I will provide instructions later.

Kind regards,
Evaldas Jocys

**Comment by freecodecampster** [comment]
Hi Evaldas,

Thanks for the quick response. The issue occurred because I couldn't see how to uninstall HID guardian in x360ce UI so I uninstalled it using device manager. On reboot that's when I could no longer use any HID devices even in safe mode. 

The fix was to attach my Windows SSD to a virtual machine VMware in my case. I could then use a mouse and keyboard again. Looking under device manager under HID I could see they all had warning symbols.  Using x360ce and checking documentation I discovered the button to uninstall HID GUARDIAN was buried under some tab and due to some GUI glitch was just a very tiny button. After pressing uninstall in device manager the warning symbols were gone and reconnecting the SSD to my computer I could use HID devices again. This just left Windows activation issues which I just about managed to fix. 

It is a really good idea to add warnings about HID GUARDIAN as it can really mess things up. Maybe even have its own special tab so uninstall is easy to find. 

Kind regards,
Dav

**Comment by EJocys** [comment]
@freecodecampster 

Tank you for sharing experience with HID Guardian. I will take it into account and remove simple HID Guardian [Install] button and redirect user to dedicated form with all warnings.

Kind Regards,
Evaldas Jocys

**Comment by EJocys** [comment]
I've added instructions on how to remove HID Guardian when Keyboard and Mouse is not working i.e. by booting into Windows Recovery Console and opening C:\Windows\System32\config\SYSTEM registry hive file manually inside Registry Editor. Also, registry file could be modified on another computer and then returned back.

https://github.com/x360ce/x360ce/wiki/HID-Guardian

**Comment by Kasun-Aka** [comment]
I also went wrong after install HID Guardian. My keyborard and mouse both not worked. But I entered to Windows recovery options menu (windows 7). Then my mouse and keyboard started working. So I choose restore option and selected the time before to install HID Guardian. Then my PC worked normally. You can search tutorial for windows recovery using youtube. Good luck

**Comment by R4to0** [comment]
I also have the same issue with latest release binary. It does make every usb port to disconnect and then reconnect, sometimes at random times, sometimes a few seconds after I open the application. If I'm playing from my external disks these instantly crashes.

**Comment by EvoJocys** [comment]
This is a HID Guardian problem. HID Guardian was discontinued by its author in 2023 and x360ce 4.17.50 marked it obsolete; keyboards, mice, audio devices and USB devices dropping out while it was installed were its known side effects. Uninstall it from Options → HID Guardian → Uninstall.

To hide a real controller from games, use HidHide instead: Options → HID Hide shows the driver state and opens its client; there, add x360ce.exe to the application list and tick the controller under Devices. Doing this from inside x360ce is tracked in #1389.

Closing as part of a tracker clean-up after the 4.21 release.

**Comment by EvoJocys** [comment]
To be precise about the closing note above: this report does not name HID Guardian, so the attribution is a judgement from the symptoms, which match the device hiding and rescans that 4.14 to 4.16 ran automatically for it. If HID Guardian was never installed and this still happens on 4.21, reopen this issue with the version and what you see; that would be a different defect and worth looking at.
